aiocache/base.py

```diff
@@ -3,16 +3,21 @@
 import logging
 import os
 import time
+from abc import abstractmethod
 from enum import Enum
 from types import TracebackType
-from typing import Callable, Optional, Set, Type
+from typing import Callable, Generic, List, Optional, Set, TYPE_CHECKING, Type, TypeVar, Union
 
-from aiocache import serializers
+if TYPE_CHECKING:
+    from aiocache.plugins import BasePlugin
+    from aiocache.serializers import BaseSerializer
+from aiocache.serializers import StringSerializer
 
 
 logger = logging.getLogger(__name__)
 
 SENTINEL = object()
+CacheKeyType = TypeVar("CacheKeyType", bound=Union[str, bytes])
```

Comment:
I think it's totally up to the class what type they use for cache keys, so I'd leave this unbounded.

Maybe some special cache uses `int` after producing some kind of hash?

Reply:
OK, now that we use the `_str_build_key()` helper, this bound is more restrictive than helpful.